src/org/mozilla/javascript/TokenStream.java

```diff
@@ -1115,6 +1115,13 @@ final int getToken() throws IOException {
                 String str = getStringFromBuffer();
                 this.string = (String) allStrings.intern(str);
                 return Token.STRING;
+            } else if (c == '#'
```

Comment:
i think we can have a plain if here like for all the other cases in the method if the block before was left with a return